### ISSUE TITLE
Barcharts with `scale.zero=false` broken

### ISSUE THREAD

**Issue by domoritz** [vega/vega-lite#1505]
Reported by @shrjain

```json
{
  "description": "A simple bar chart with embedded data.",
  "data": {
    "values": [
      {"a": "A","b": 28}, {"a": "B","b": 55}, {"a": "C","b": 43},
      {"a": "D","b": 91}, {"a": "E","b": 81}, {"a": "F","b": 53},
      {"a": "G","b": 19}, {"a": "H","b": 87}, {"a": "I","b": 52}
    ]
  },
  "mark": "bar",
  "encoding": {
    "x": {"field": "a", "type": "ordinal"},
    "y": {"field": "b", "type": "quantitative", "scale": {"zero": false}}
  }
}

```

**Comment by domoritz** [comment]
in https://github.com/vega/vega-lite/blob/master/src/compile/mark/bar.ts#L159, we shoulnd't set 0 but instead use the chart height.

**Comment by kanitw** [comment]
Should be addressed in #1507 ... need unit test to prevent regression
